sdk/servicebus/service-bus/samples/typescript/src/browseMessages.ts

```diff
@@ -17,30 +17,33 @@ import * as dotenv from "dotenv";
 dotenv.config();
 
 // Define connection string and related Service Bus entity names here
-const connectionString = process.env.SERVICE_BUS_CONNECTION_STRING || "<connection string>";
+const connectionString =
+  process.env.SERVICE_BUS_CONNECTION_STRING || "<connection string>";
 const queueName = process.env.QUEUE_NAME || "<queue name>";
 
 export async function main() {
-  const sbClient = ServiceBusClient.createFromConnectionString(connectionString);
+  const sbClient = new ServiceBusClient(connectionString);
 
-  // If using Topics & Subscription, use createSubscriptionClient to peek from the subscription
-  const queueClient = sbClient.createQueueClient(queueName);
+  // If receiving from a subscription you can use the getReceiver(topic, subscription) overload
+  // In this case since we're using `diagnostics.peek()` that any lock mode will work.
+  // `diagnostics.peek()` does not lock messages in either lock mode.
+  const queueReceiver = sbClient.getReceiver(queueName, "receiveAndDelete");
 
   try {
     for (let i = 0; i < 20; i++) {
-      const messages = await queueClient.peek();
+      const messages = await queueReceiver.diagnostics.peek();
```

Comment:
Unrelated to the PR...
Did we finalize `diagnostics`?

Reply:
Not for all other languages but, for JS, we did say in the arch meeting that we'd try it as an experiment. (naming aside).